### ISSUE TITLE
Youtube Creates Insane Stacked Lag

### ISSUE THREAD

**Issue by calicocookiecutter** [BrowserWorks/waterfox#3632]
### What happened?

Hi there, basically, the longer I use YouTube, the laggier waterfox gets, the video will rarely freeze but usually keeps playing normally. Things like pausing or hovering with a mouse to see the display (timeline, settings, etc) gets worse and worse until it doesn't respond at all
This lag eventually spreads to the start menu, I can pull it up with the WIN key, but no mouse or keyboard inputs register.
This means I can't close the window at all, this leads me to force shutting down my computer using the power button.
Other programs run fine during this? EXCEPT once when I realized waterfox had 'frozen' (again, video still plays), at some point it had corrupted my Clip Studio Paint file. There's a known bug that rarely a file can corrupt before saving and/or quitting, which turns it into a .csp file that isn't recognized as a .csp file anymore by CSP, which is exactly what happened.

The onset of the lag usually takes an hour or two but ranges, sometimes it can occur really quickly, skipping straight to 'freezing'.
I used to put up with it, even if it was super annoying, but I just lost hours of work to this. I have NEVER had any issues with CSP up till now, I've been using it for 3 years.
It has persisted for many editions.
Considering my CSP file corrupted and the video still plays whilst everything else has gone to shit, I'm thinking it may be like, a computational demand issue, thing? Like my computer is putting in too much effort and piling up too much junk data from the youtube window that other things just cant work. I've put my PC specs in-case this is the issue.

TL;DR
- Youtube will get increasingly worse input delay before becoming unresponsive to everything, video will still play
- This spreads to the search function on windows, WIN key can pull it up, but no other inputs work
- Refreshing does not fix the issue
- Once 'frozen', the computer must be externally powered off in order to regain control
- Other applications (only tested with CSP) work 'fine'
- Very likely corrupted my CSP file
- Persisted for many editions

PC Specs:
- 12th gen intel corei5-12400F
- nvidia geforce rtx 3060
- 48gbs RAM (3200 MT/s, ddr4)
- Two monitors (gives context)

Please fix this issue g's, I'm seriously considering going back to chrome because of this :((

### Reproducible?

- [ ] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

G6

### What platform are you seeing the problem on?

Windows

### Relevant log output

```shell

```

**Comment by SavageTheUnicorn** [comment]
I experience the same freezing issues more often when multiple tabs of youtube videos are open. For example; I have 8 tabs (none have began loading the video at all yet, I did this intentionally to save on resources) and while watching a video I'd estimate once every 15-30 seconds the entire video playback freezes EXCEPT it doesn't. Only the audio continues playing, but waiting a few seconds and interacting with the window in any way seems to get it to unstick kind of.

(I think this issue, for me atleast. was fixed and then REINTRODUCED in the latest update. One of the last 2-3 updates had solved this for me)

**Comment by calicocookiecutter** [comment]
> I experience the same freezing issues more often when multiple tabs of youtube videos are open. For example; I have 8 tabs (none have began loading the video at all yet, I did this intentionally to save on resources) and while watching a video I'd estimate once every 15-30 seconds the entire video playback freezes EXCEPT it doesn't. Only the audio continues playing, but waiting a few seconds and interacting with the window in any way seems to get it to unstick kind of.
> 
> (I think this issue, for me atleast. was fixed and then REINTRODUCED in the latest update. One of the last 2-3 updates had solved this for me)

oh weird, i think it happens more often for me when i have two windows open with 7+ tabs total, but only one of those is youtube so

**Comment by Squall-Leonhart** [comment]
There is an active memory leak bug involving Youtube on the mozilla tracker, the issue is likely from that.

**Comment by SavageTheUnicorn** [comment]
> There is an active memory leak bug involving Youtube on the mozilla tracker, the issue is likely from that.

That's very possible. For now I've turned off hardware acceleration and seems to have solved it though it's only been about 8 hours.

After extended testing, this is a viable solution for now (for me).

After even more testing, it was a false positive. It seems to take a bit longer to create the lag, but it still happens sadly.

**Comment by IdrisQe** [comment]
Hi, on 6.5.2 here, and I can confirm that turning off Hardware Acceleration does _not_ guarenteed fix this. I already had it disabled before updating to 6.5.X and Youtube gets compoundingly more laggy the longer I have it open for.

I might try disabling "media.hardware-video-decoding.enabled" in about:config and see if that helps, though I have a feeling that will just make things worse. If I do remember to do that tomorrow, I'll report back and edit this comment as to whether it helped or not.

**Comment by calicocookiecutter** [comment]
Update, you can see the lag progress in real time when scrolling through the youtube homepage, so its not specific to media players it seems?

**Comment by Kol9yN** [comment]
@IdrisQe Hello, any luck with testing?
Got same problem, and need to restart WF after some time

**Comment by calicocookiecutter** [comment]
Hi again, another update
Making a big queue pretty makes immediate lag, so maybe that combined with the scrolling thing i mentioned earlier, it has something to do with the thumbnail playbacks? Dunno, i know jack abt software lol

**Comment by Squall-Leonhart** [comment]
Pick your poison https://bugzilla.mozilla.org/buglist.cgi?quicksearch=youtube+memory

**Comment by IdrisQe** [comment]
> [@IdrisQe](https://github.com/IdrisQe) Hello, any luck with testing? Got same problem, and need to restart WF after some time

Sorry for the lack of response, been very distracted lately getting ready for, and subsequently going on, an international trip to see my partner. It did seem to help a lot, though that may have also been a subsequent Waterfox update, or perhaps my reinstall of Waterfox on the laptop I'm using while travelling, or the laptop itself having an embarassingly better CPU than my Desktop, or any other number of factors.

So the answer is: Maybe? Sorry I don't have anything concrete.

**Comment by SavageTheUnicorn** [comment]
Latest waterfox version seems to have fixed this for me., updated less than 24hrs ago. Will update this comment with an edit if anything arises.

**Comment by SpoopyTim** [comment]
I thought I was going crazy. I can confirm this happens to me so often its actually insane. I have to restart waterfox at least 10 times a day to fix the built up lag. Tried disabling hardware acceleration, disabling dns over https (because i already use a secure dns resolver), messing with debug settings and the like. It seems that cpu usage or ram usage is not unusually high for the things I have open. Unloading tabs with [UnloadTabs](https://addons.mozilla.org/en-GB/firefox/addon/unload-tabs/) can prolong the inevitable restart but does not fix the issue on its own. I have yet to test a browser run without loading youtube or a clean profile so that will be next when I can.

**Comment by Squall-Leonhart** [comment]
This is fixed by https://bugzilla.mozilla.org/show_bug.cgi?id=1939295 upstream

**Comment by SavageTheUnicorn** [comment]
> This is fixed by https://bugzilla.mozilla.org/show_bug.cgi?id=1939295 upstream

Not to be a total github noob but... has this been merged for waterfox yet? dev version only? 6.5.4 came out 2 months ago so

**Comment by Xavron** [comment]
Maybe its only fixed for Youtube and nothing else? But, I get this with Waterfox, and still even the latest Firefox (currently 136.0.2) on non-Youtube video streaming. I don't think I've seen it on Youtube, but I don't use Youtube all that much.

I tested it the other day as showing up even on Firefox 130 although I don't remember it being a problem back then (even if its about when it started for me, eg 126 breaks an extension I use, I'd have to try on 127-129 which is annoying).

It also doesn't appear to affect all streaming, just certain ones.

I'm not spending the time to debug it. Just adding this. Surely it will be fixed someday.

That link makes it seems like its fixed in a release build already but that's not what I'm seeing.

I also see it on two fully different computers. It could be a result from the website side but I don't really see that here as a reason for a cause.

The actual problem seems to be underestimated. I mostly expect it will be fixed in 6+ months. Maybe years. Some issues take Firefox devs that long. If its fixed in 127, 128 or something, I'll be pleasantly surprised.

**Comment by Squall-Leonhart** [comment]
> I tested it the other day as showing up even on Firefox 130 although I don't remember it being a problem back then (even if its about when it started for me, eg 126 breaks an extension I use, I'd have to try on 127-129 which is annoying).

the fix wasn't in 130, but is in esr 128.

> Not to be a total github noob but... has this been merged for waterfox yet? dev version only? 6.5.4 came out 2 months ago so

Waiting on Alex to bring in the latest ESR fixes.

**Comment by Xavron** [comment]
128 is before 130 though, so no, its not working except _maybe_ for YouTube. The issue will _still_ exist based on that issue supposedly fixed in 128.

This doesn't inflict Firefox on Linux (just tested enough). So, it is a Windows only Firefox based code, or side issue affecting the code on Windows. This is looking to be my solution for the time being, so I'm done posting here.

Based on your reply, it continues to be underestimated by people. This affects more than just YouTube and is still found in latest Firefox release. I'd be surprised if what is pulled in (with current context) that it will actually fully solve it. :'(

It is possible there's multiple issues pounding on this. Just as its possible the "fix" wasn't a good one. (I don't see anything in that code that is applied to Windows only so its a poor fix imo, if one can call it a fix. Although, only quickly looked at it. But, I'm done here. Not wasting any further time on this as a solution exists that is good enough for me.)

Hope it goes well :)

**Comment by Squall-Leonhart** [comment]
> 128 is before 130 though, so no, its not working except _maybe_ for YouTube. The issue will _still_ exist based on that issue supposedly fixed in 128.

You misunderstand versioning

128ESR is not on a tree that 130 would take fixes on.

The first point in which the issue is fixed for main branch Firefox is 136.xx, but yes there are still some other YouTube specific issues open, one being particular to the subscriptions page, another to live stream chat.

There are also extension specific YouTube and twitch issues active

**Comment by Xavron** [comment]
> > 128 is before 130 though, so no, its not working except _maybe_ for YouTube. The issue will _still_ exist based on that issue supposedly fixed in 128.
> 
> You misunderstand versioning
> 
> 128ESR is not on a tree that 130 would take fixes on.
> 
> The first point in which the issue is fixed for main branch Firefox is 136.xx, but yes there are still some other YouTube specific issues open, one being particular to the subscriptions page, another to live stream chat.
> 
> There are also extension specific YouTube and twitch issues active

**Edit: As it is for me, as I'm seeing it, looks fixed on Firefox 137.0 on Windows. WaterFox should get it whenever it does.**

Thank you for trying to explain, but I'm not sure if I'm not understanding something here correctly, or am being misunderstood :)

I'm seeing this not with YouTube on Windows. Nothing to do with YouTube or anything else you mention in my case. _But, it is the exact same problem of this issue._

Inevitably, it will be fixed it time anyway regardless of 13x, ESR, or anything. I don't really want to keep going on this. For now, using via Linux solves it for me and that's fine.

Thanks though. I do appreciate it :)

**Comment by Squall-Leonhart** [comment]
still a bug (but its upstream)
